apisix/cmd/ops.lua

```diff
@@ -0,0 +1,294 @@
+
+    if util.is_32bit_arch() then
+        sys_conf["worker_rlimit_core"] = "4G"
+    else
+        sys_conf["worker_rlimit_core"] = "16G"
+    end
+
+    for k,v in pairs(yaml_conf.apisix) do
+        sys_conf[k] = v
+    end
+
+    for k,v in pairs(yaml_conf.nginx_config) do
+        sys_conf[k] = v
+    end
+
+    local wrn = sys_conf["worker_rlimit_nofile"]
+    local wc = sys_conf["event"]["worker_connections"]
+    if not wrn or wrn <= wc then
+        -- ensure the number of fds is slightly larger than the number of conn
+        sys_conf["worker_rlimit_nofile"] = wc + 128
+    end
+
+    if sys_conf["enable_dev_mode"] == true then
+        sys_conf["worker_processes"] = 1
+        sys_conf["enable_reuseport"] = false
+
+    elseif tonumber(sys_conf["worker_processes"]) == nil then
+        sys_conf["worker_processes"] = "auto"
+    end
+
+    local dns_resolver = sys_conf["dns_resolver"]
+    if not dns_resolver or #dns_resolver == 0 then
+        local dns_addrs, err = util.local_dns_resolver("/etc/resolv.conf")
+        if not dns_addrs then
+            return util.die("failed to import local DNS: ", err)
+        end
+
+        if #dns_addrs == 0 then
+            return util.die("local DNS is empty")
+        end
+
+        sys_conf["dns_resolver"] = dns_addrs
+    end
+
+    local conf_render = template.compile(ngx_tpl)
+    local ngxconf = conf_render(sys_conf)
+
+    local ok, err = file.write_file(env.apisix_home .. "/conf/nginx.conf",
+                                    ngxconf)
+    if not ok then
+        return util.die("failed to update nginx.conf: ", err)
+    end
+
+    local op_ver = get_openresty_version()
+    if op_ver == nil then
+        return util.die("can not find openresty\n")
+    end
+
+    local need_ver = "1.15.8"
+    if not check_version(op_ver, need_ver) then
+        return util.die("openresty version must >=", need_ver, " current ", op_ver, "\n")
+    end
+end
+
+
+function _M.start(...)
+    -- check running
+    local pid = file.read_file(env.pid_path)
+    if pid then
+        local hd = popen("lsof -p " .. pid)
+        local res = hd:read("*a")
+        if res and res ~= "" then
+            print("APISIX is running...")
+            return nil
+        end
+    end
+
+    _M.init(...)
+    etcd.init(...)
+
+    execute(openresty_args)
+end
+
+
+function _M.stop()
+    execute(openresty_args .. [[ -s stop]])
+end
+
+
+function _M.restart()
+    _M.stop()
+    _M.start()
+end
+
+
+function _M.reload()
+    -- reinit nginx.conf
+    _M.init()
+
+    local test_cmd = openresty_args .. [[ -t -q ]]
+    -- When success,
+    -- On linux, os.execute returns 0,
+    -- On macos, os.execute returns 3 values: true, exit, 0,
+    -- and we need the first.
+    local test_ret = execute(test_cmd)
+    if test_ret == 0 or test_ret == true then
+        local cmd = openresty_args .. [[ -s reload]]
+        execute(cmd)
+        return
+    end
+
+    util.die("test openresty failed")
+end
+
+
+return _M
```

Comment:
Missing `_M.help` and `_M.version`?

Reply:
The help and version sub command locate at `bin/apisix` file since the don't belong to the scope of `ops`.